docs/DeveloperGuide.adoc

```diff
@@ -142,71 +155,356 @@ image::StorageClassDiagram.png[]
+5. `EditUserCommand` then calls `execute()` which first gets the existing user in Model as `userToEdit`. Then it creates a new
+`editedUser` based on `EditUserDescriptor`. It will then update the current `userToEdit` in Model with `editedUser`. Afterwards,
+a `CommandResult` is returned to Logic with data and feedback to be displayed to the user.
+6. Feedback acknowledge is displayed by `ResultDisplay`. User profile changes are displayed by JavaFx `TableView`.
+
+The following sequence diagram shows how the `me` feature allows user to edit the user profile:
+
+image::MeSequenceDiagram.png[]
+
+===== Design Considerations
+
+*Aspect: Whether `EditUserCommand` should extend `EditCommand`*
+
+* ** Alternative 1 (current choice):** `EditUserCommand` does not extend `EditCommand`, but extends `Command`
+This design is chosen because while `EditCommand` takes into account the item index as all items are stored in a list in Model, `EditUserCommand`
+only concerns with a single `Person` who is the sole user.
+
+** Pros: Reduces unnecessary overhead for `EditUserCommand`.
+** Cons: Unable to exploit polymorphism if there is similarity with the `EditCommand`.
+
+* **Alternative 2: `EditUserCommand` extends `EditCommand`**
+
+** Pros: Better utilise polymorphism and perhaps can be more intuitive as it is also a command to edit.
+
+** Cons: Does not treat it as an entirely separate command with a distinct keyword `me` which is more intuitive for the user.
+
+*Aspect: Whether to have both `AddUserCommand` and `EditUserCommand`*
+
+* *Alternative 1 (current choice):* A default user data is initialized and displayed at first start-up. User can update it afterwards.
+This design is chosen because `EditUserCommand` only concerns with a single `Person` object in the Model as the sole user. Hence
+there is no need for `AddUserCommand` as `EditUserCommand` when executed always creates a new `Person` object to replace the
+existing one and update the Model and Ui accordingly.
+
+** Pros: Reduces unnecessary code duplication.
+** Cons: User may expect to have `add` command intuitively.
+
+* Alternative 2: Have both `AddUserCommand` and `EditUserCommand`
+
+** Pros: User can intuitively treat `add` as adding in a new `user` and `edit` as just modifying an existing `user`.
+** Cons: There will be code duplication and the one single user logic is not fully utilised to reduce code.
+
+==== Store and Display Motivational Quotes
+This allows the user to input and store motivational quotes. They are then able to display them at the user profile pane.
+
+==== Set Personal Reminders and Take Notes
+This allows the user to set reminders as well as taking notes for him/herself to further customize his/her interaction
+with the app.
+
+==== Change Background and Font Color
+This allows the user to change the background and font color using rgb or hex color codes.
```

Comment:
Do you think listing these sections contribute to a reader's understanding of the implementation for these features?